pkg/catalog/catalog_test.go

```diff
@@ -92,45 +92,91 @@ func TestVersionExist(t *testing.T) {
 
 func TestList(t *testing.T) {
 	type args struct {
-		name      string
-		namespace string
-		tags      []string
+		name               string
+		namespace          string
+		tags               []string
+		currentNSImageName string
+		currentNamespace   string
+		currentNSImageTags []string
 	}
 	tests := []struct {
-		name     string
-		args     args
-		wantErr  bool
-		wantTags []string
+		name    string
+		args    args
+		wantErr bool
+		wantOp  []CatalogImage // Expect 1 duplicate because of 2 calls(1 for current namespace and another for openshift namespace) to client.GetImageStreams per call to list
```

Comment:
@kadel Is there a better way of doing this test
i.e, https://github.com/redhat-developer/odo/pull/618/files#diff-1dfbb1796e8d3ca31e26fe9d003bb488R188 for particular parameters instead of whole list?

Reply:
I think that this is ok, but the duplicate stuff is a bit weird